changelog.md

```diff
@@ -1,5 +1,12 @@
 # Changelog
 
+## NEXT
+
+- Fixed a bug that would allow the addition of b--a and b->a edges to a graph with an existing undirected a--b edge in
+  the `cai_causal_graph.causal_graph.CausalGraph._set_edge` method.
```

Comment:
Would also say the `.add_edge` method here instead of the `._set_edge` method, even though the fix is for the latter.

The main reason for that is that the changelog is customer/public facing and they won't be exposed to protected/private methods, so better to talk about the public methods where this bug was observed.